### PR TITLE
Remove the renewal lock file upon uninstall

### PR DESCRIPTION
**Pull request by flo-renaud**
Make sure that the file /var/run/ipa/renewal.lock is deleted upon
uninstallation, in order to avoid subsequent installation issues.

Part of the refactoring effort, certificates sub-effort.

https://fedorahosted.org/freeipa/ticket/6433

### CODE REVIEW

**Comment by jcholast** [comment]
The file is owned by the server, not the client, so it should be deleted in `ipa-server-install --uninstall`, not in `ipa-client-install --uninstall`.

**Comment by flo-renaud** [comment]
You are right, I updated the PR to put the code at the end of server uninstallation.

**Comment by frasertweedale** [comment]
Works as expected.

File: ipapython/certmonger.py [hunk 1]
```diff
@@ -307,17 +307,30 @@ def request_and_wait_for_cert(
     The method also waits for the certificate to be available.
     """
     reqId = request_cert(nssdb, nickname, subject, principal,
-                         passwd_fname, dns, ca, profile,
-                         pre_command, post_command)
-    state = wait_for_request(reqId, timeout=60)
+                         passwd_fname, dns, ca, profile)
+    state = wait_for_request(reqId, timeout=180)
```
Comment (jcholast) [inline ipapython/certmonger.py]: Could you please put the timeout increase into a separate commit?

File: ipaserver/install/server/install.py [hunk 2]
```diff
@@ -1130,6 +1130,12 @@ def uninstall(installer):
                           ' # getcert stop-tracking -i <request_id>\n'
                           'for each id in: %s' % ', '.join(ids))
 
+    # Remove the cert renewal lock file
+    try:
+        os.remove(paths.IPA_RENEWAL_LOCK)
+    except Exception:
+        pass
```
Comment (jcholast) [inline ipaserver/install/server/install.py]: This would be better (off the top of my head):
```python
    try:
        os.remove(paths.IPA_RENEWAL_LOCK)
    except OSError as e:
        if e.errno != errno.ENOENT:
            root_logger.warning("Failed to remove %s: %s", paths.IPA_RENEWAL_LOCK, e)
```

File: ipapython/certmonger.py [hunk 3]
```diff
@@ -307,17 +307,30 @@ def request_and_wait_for_cert(
...
     ca_error = get_request_value(reqId, 'ca-error')
     if state != 'MONITORING' or ca_error:
         raise RuntimeError("Certificate issuance failed")
+
+    # Add presave and postsave commands
+    # This is not done earlier to avoid running the cmds
+    # during the initial request
```
Comment (jcholast) [inline ipapython/certmonger.py]: Instead of this, I would rather not restart the services in `renew_ra_cert`, `restart_httpd` and `restart_dirsrv` if they were not previously running.

File: ipapython/certmonger.py (hunk 1 above)
Comment (jcholast) [inline ipapython/certmonger.py]: Maybe it might not even be necessary with the other fixes.

File: ipapython/certmonger.py (hunk 3 above)
Comment (jcholast) [inline ipapython/certmonger.py]: Speaking of which, I don't think the restart in `renew_ra_cert` is necessary at all - httpd itself does not use the RA cert in any way, it is used only in the framework.

**Comment by flo-renaud** [comment]
Hi,
I implemented @jcholast suggestions and finally found the origin of the lock.

**Comment by mbasti-rh** [comment]
Works for me on both domain levels, I'd ACK this if nobody is against

**Comment by mbasti-rh** [comment]
Fixed upstream
master:
https://fedorahosted.org/freeipa/changeset/198cd5fab3937fd8948bea4b4949e30db4e490a4
